pkg/koordlet/util/perf_group/perf_group_linux.go

```diff
@@ -270,23 +244,13 @@ func GetAndStartPerfGroupCollectorOnContainer(cgroupFile *os.File, cpus []int, e
 
 func GetContainerPerfResult(collector *PerfGroupCollector) (map[string]float64, error) {
 	var err error
-	var mu sync.Mutex
-	var wg sync.WaitGroup
-	wg.Add(len(collector.cpus))
 	for _, cpu := range collector.cpus {
-		go func(cpu int) {
-			defer wg.Done()
-			if pcInf, ok := collector.perfCollectors.Load(cpu); ok {
-				pc := pcInf.(*perfCollector)
-				if err = pc.collect(collector.valueCh); err != nil {
-					mu.Lock()
-					err = multierr.Append(err, err)
-					mu.Unlock()
-				}
+		if pc, ok := collector.perfCollectors[cpu]; ok {
```

Comment:
> The code to reuse `attrMap` looks good to me. Anyway, what is the reason to delete the usage of `WaitGroup?` for all cpus?

From `perf trace` stat, concurrent syscall will cause queue, `perf_event_open` will cost more than 10s(maximum)

Reply:
> > The code to reuse `attrMap` looks good to me. Anyway, what is the reason to delete the usage of `WaitGroup?` for all cpus?
> 
> From `perf trace` stat, concurrent syscall will cause queue, `perf_event_open` will cost more than 10s(maximum)

The reason at first to use concurrent syscall is to assure the collection of CPI data for one container on all cpus starts at the same time. Otherwise it will cause precision lost.

Now that it shows that concurrent syscall cost more than 10s overhead at maximum, I think it is a good tradeoff to give the concurrent way up and accept the for-loop way. 

cc @zwzhang0107 @saintube

Reply:
/lgtm

Reply:
/lgtm